File: C++/LabWork#2/ExpressionOperations.cpp
#include "ExpressionOperations.h"
#include <algorithm>

ExpressionOperations::ExpressionOperations() {
  text = "";
}

ExpressionOperations::ExpressionOperations(std::string input) {
  text = input;
}

ExpressionOperations::ExpressionOperations(const ExpressionOperations& obj) {
  text = obj.text;
}

ExpressionOperations::~ExpressionOperations() {
}

std::string ExpressionOperations::expressionToUpper() {
  std::transform(text.begin(), text.end(), text.begin(), ::toupper);
  return text;
}

std::string ExpressionOperations::getText() {
  return text;
}

int ExpressionOperations::calculateExpressionLength() {
  return text.length();
}

std::string ExpressionOperations::replaceChars(char oldChar, char newChar) {
  std::string result = text;
  std::replace(result.begin(), result.end(), oldChar, newChar);
  return result;
}

File: C#/LabWork#2/ExpressionOperations.cs
using System;

namespace LabWork_2
{
  public class ExpressionOperations
  {
    private string text;

    public ExpressionOperations()
    {
      text = "";
    }

    public ExpressionOperations(string input)
    {
      text = input;
    }

    public ExpressionOperations(ExpressionOperations anotherObj)
    {
      text = anotherObj.text;
    }

    ~ExpressionOperations()
    {
    }

    public string expressionToUpper()
    {
      return text.ToUpper();
    }

    public string GetText()
    {
      return expressionToUpper();
    }

    public int CalculateExpressionLength()
    {
      return text.Length;
    }

    public string ReplaceChars(char oldChar, char newChar)
    {
      return text.Replace(oldChar, newChar).ToUpper();
    }
  }
}
